contracts/pair_stable_bluna/src/contract.rs

```diff
@@ -536,8 +538,40 @@ pub fn provide_liquidity(
             return Err(ContractError::LiquidityAmountTooSmall {});
         }
 
+        // Get fee info from the factory
+        let fee_info = query_fee_info(
+            &deps.querier,
+            &config.factory_addr,
+            config.pair_info.pair_type.clone(),
+        )
+        .unwrap_or_default(); // There may no fee exist thus 0 is a default fee.
+
+        let fee = fee_info.total_fee_rate / Uint128::from(2u8);
```

Comment:
Do not use magic numbers, please. If it possible use constant or add some comment